hi-ml/testhiml/testhiml/test_run_ml.py

```diff
@@ -1,21 +1,37 @@
+import torch
 import shutil
-from pathlib import Path
-
 import pytest
+
+from pathlib import Path
 from typing import Generator
 from unittest.mock import MagicMock, Mock, patch
 
+from azureml.core import Workspace
 from health_ml.configs.hello_world import HelloWorld  # type: ignore
 from health_ml.experiment_config import ExperimentConfig
 from health_ml.lightning_container import LightningContainer
 from health_ml.run_ml import MLRunner
 from health_ml.utils.common_utils import is_gpu_available
-from health_azure.utils import is_global_rank_zero
-
+from health_azure.utils import is_global_rank_zero, create_aml_run_object
+from health_azure.utils import WORKSPACE_CONFIG_JSON, check_config_json, get_workspace
+from testazure.utils_testazure import DEFAULT_WORKSPACE
+from testhiml.utils.fixed_paths_for_tests import full_test_data_path
+from testazure.utils_testazure import get_shared_config_json
 
 no_gpu = not is_gpu_available()
 
 
+def _get_workspace(tmp_path: Path) -> Workspace:
```

Comment:
I'm not sure why this is necessary? lines 30-31 work on a local machine too

Reply:
Definitely, but you need to set the environment variables for the service principal. Not everyone has these set especially new joiners, whereas everyone has a config.json at the repo root for other purposes.  It's just to make it easier to debug the tests without having to go through all the service principal setup

Reply:
We have code like that already. For unittests, just use `DEFAULT_WORKSPACE.workspace`, this has all the smarts already built in.